### PR TITLE
PLANNER-2389 Upgrade Quarkus to 1.13

### PR DESCRIPTION
**Pull request by rsynek**
### JIRA
https://issues.redhat.com/browse/PLANNER-2389

### Referenced pull requests
https://github.com/kiegroup/optaplanner/pull/1254
https://github.com/kiegroup/optaplanner-quickstarts/pull/93
https://github.com/kiegroup/optaweb-employee-rostering/pull/602
https://github.com/kiegroup/optaweb-vehicle-routing/pull/471

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
</details>

### CODE REVIEW

**Comment by rsynek** [comment]
Jenkins run native

File: build/optaplanner-build-parent/pom.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
     <version.com.fasterxml.jackson.core>2.12.1</version.com.fasterxml.jackson.core>
     <version.com.h2database>1.3.173</version.com.h2database>
     <version.com.thoughtworks.xstream>1.4.16</version.com.thoughtworks.xstream>
-    <version.io.quarkus>1.11.0.Final</version.io.quarkus>
+    <version.io.quarkus>1.13.0.Final</version.io.quarkus>
```
Comment (triceo) [inline build/optaplanner-build-parent/pom.xml:35]: There is a whole bunch of other dependencies in this POM, which may be overriding their versions coming from master.
Did you also check if those need to be bumped too?

Comment (ge0ffrey) [reply build/optaplanner-build-parent/pom.xml:35]: In the long run, we're thinking of starting by importing a quarkus bom (question is which one, not the platform bom of course as that contains optaplanner), so we don't get to decide our hibernate version any more, nor have to keep it in sync.

**Review by ge0ffrey — APPROVED** [review]

**Review by triceo — APPROVED** [review]
Good when green.

**Comment by rsynek** [comment]
Waiting for all Linux builds on the entire ensemble of PRs before merging.

**Comment by rsynek** [comment]
Could not reproduce the issue locally, nor it shows on the other CI. Either it's a random glitch in Jenkins or some environment-specific issue.

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
Jenkins run native

**Comment by rsynek** [comment]
Let's try to reproduce the Jenkins-specific issue with Quarkus 1.13.0.Final to see if there is a regression in one of the bugfix releases.

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
Jenkins run native

**Pull request merged**